File: data_structures/q6/solution.cpp
#include<iostream>
#include<vector>
using namespace std;

void reverseArray(vector<int>& arr, int n){
    int left = 0, right = n-1;
    while(left < right){
        int temp = arr[left];
        arr[left] = arr[right];
        arr[right] = temp;
        left++;
        right--; 
    }
}

int main(){
    vector<int> arr = {10,9,8,7,6,5,4,3,2,1};

    reverseArray(arr , arr.size());

    cout << "The reversed array is: ";
    for(int i = 0; i < arr.size(); i++){
        cout << arr[i] << " ";
    }
    return 0;
}

File: data_structures/q6/solution.py
def reverseArray(arr,n):
    left = 0
    right = n-1
    temp = 0
    while(left < right):
        temp = arr[left]
        arr[left] = arr[right]
        arr[right] = temp
        left+=1
        right-=1
    return arr

main_arr = [10,9,8,7,6,5,4,3,2,1]
print(reverseArray(main_arr, len(main_arr)))